recipes/libtorrent/all/conanfile.py

```diff
@@ -45,126 +49,129 @@ class LibtorrentConan(ConanFile):
-        self._cmake.definitions["python-bindings"] = False
-        if self._is_msvc:
-            self._cmake.definitions["static_runtime"] = "MT" in msvc_runtime_flag(self)
-        self._cmake.configure()
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["Boost_USE_STATIC_LIBS"] = not self.options["boost"].shared
+        tc.variables["deprecated-functions"] = self.options.enable_deprecated_functions
+        tc.variables["dht"] = self.options.enable_dht
+        tc.variables["encryption"] = self.options.enable_encryption
+        tc.variables["exceptions"] = self.options.enable_exceptions
+        tc.variables["i2p"] = self.options.enable_i2p
+        tc.variables["logging"] = self.options.enable_logging
+        tc.variables["mutable-torrents"] = self.options.enable_mutable_torrents
+        tc.variables["build_tests"] = False
+        tc.variables["build_examples"] = False
+        tc.variables["build_tools"] = False
+        tc.variables["python-bindings"] = False
+        tc.variables["python-bindings"] = False
+        if is_msvc(self):
+            tc.variables["static_runtime"] = "MT" in msvc_runtime_flag(self)
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        for patch_data in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch_data)
+        apply_conandata_patches(self)
 
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "/W4", "")
-        if tools.Version(self.version) < "2.0":
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "/W4", "")
+        if Version(self.version) < "2.0":
             if self.options.enable_iconv:
                 replace = "find_public_dependency(Iconv REQUIRED)"
             else:
                 replace = "set(Iconv_FOUND OFF)"
-            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
                                   "find_public_dependency(Iconv)",
                                   replace)
             if self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libstdc++":
                 # https://github.com/arvidn/libtorrent/issues/3557
-                tools.replace_in_file(os.path.join(self._source_subfolder, "include", "libtorrent", "file_storage.hpp"),
+                replace_in_file(self, os.path.join(self.source_folder, "include", "libtorrent", "file_storage.hpp"),
                                       "file_entry& operator=(file_entry&&) & noexcept = default;",
                                       "file_entry& operator=(file_entry&&) & = default;")
 
     def build(self):
         self._validate_dependency_graph()
```

Comment:
to move to validate() now

Reply:
@SpaceIm 
I combine `_validate_dependency_graph()` to `validate()`.
Thanks.